ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java

```diff
@@ -528,9 +531,20 @@ public void getTableList(Context ctx) {
           tables.stream().map(TableMeta::getName).collect(Collectors.toSet());
       hiveTables.stream()
           .filter(e -> !arcticTables.contains(e))
-          .forEach(e -> tables.add(new TableMeta(e, TableMeta.TableType.HIVE.toString())));
+          .forEach(e -> lackHiveTables.add(new TableMeta(e, TableMeta.TableType.HIVE.toString())));
     }
-    tables.sort((Comparator.comparing(TableMeta::getName)));
+    // sort by format
+    tables.sort(
```

Comment:
We can sort within the stream API when the list of `tables` is built, then the codes can be significantly simplified, like 
```java
 List<TableMeta> tables =
        tableService.listTables(catalog, db).stream()
            .map(
                idWithFormat ->
                    new TableMeta(
                        idWithFormat.getIdentifier().getTableName(),
                        formatToType.apply(idWithFormat.getTableFormat())))
            .sorted(xxxxx) // changed here
            .collect(Collectors.toList());
```
```java
      hiveTables.stream()
          .filter(e -> !arcticTables.contains(e))
          .sorted() // changed here
          .forEach(e -> tables.add(new TableMeta(e, TableMeta.TableType.HIVE.toString())));

Reply:
done